common/flatpak-dir.c

```diff
@@ -9724,13 +9724,36 @@ flatpak_dir_update_summary (FlatpakDir   *self,
   if (!flatpak_dir_ensure_repo (self, cancellable, error))
     return FALSE;
 
-  /* Keep a shared repo lock to avoid prunes removing objects we're relying on
-   * while generating the summary. */
-  if (!flatpak_dir_repo_lock (self, &lock, LOCK_SH, cancellable, error))
-    return FALSE;
+  if (delete)
+    {
+      g_autoptr(GError) local_error = NULL;
+      g_autoptr(GFile) summary_file = NULL;
+
+      g_debug ("Deleting summary");
+
+      summary_file = g_file_get_child (ostree_repo_get_path (self->repo), "summary");
+
+      if (!g_file_delete (summary_file, cancellable, &local_error) &&
+          !g_error_matches (local_error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND))
```

Comment:
Are we always allowed to do this? Wondering if we might get a 'permission denied' error here

Reply:
Yeah, for the system installation where we wouldn't have permission this is done by the system-helper.